File: 第1章 古算趣题/1.7 群羊逐草/群羊逐草.py
'''
程序：群羊逐草
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
def main():
    '''群羊逐草'''
    x = 1
    while True:
        if x * (1 + 1 + 1/2 + 1/4) + 1 == 100:
            print('这群羊一共有%d只' % x)
            break
        x += 1

if __name__ == '__main__':
    main()

File: 第1章 古算趣题/1.7 群羊逐草/群羊逐草.cpp
/*
程序：群羊逐草 
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
#include <bits/stdc++.h>
using namespace std;
// 群羊逐草 
int main()
{
	int x = 1;
	while (true) {
		if (x * (1 + 1 + 1/2.0 + 1/4.0) + 1 == 100) {
			cout << "这群羊一共有" << x << "只" << endl; 
			break;
		}
		x += 1;
	}
    return 0;
}
